py-ipyvuetify: new package

jeremyfix opened the pull request:
Brings in the ipyvuetify python package 
"
    Jupyter widgets based on vuetify UI components which implement Google's
    Material Design Spec with the Vue.js framework.
"

jeremyfix committed:
adds the ipyvuetify spack recipe

jeremyfix committed:
adds the ipyvuetify spack recipe

jeremyfix committed:
Merge branch 'develop' into py-ipyvuetify

adamjstewart committed:
GDAL: add v3.8.4 (#42805)

alexrichert committed:
dakota: add boost components (#42659)

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

johnwparent committed:
Seacas package: add Windows support (#42692)

adamjstewart committed:
PythonPackage: check purelib for libs/headers (#42602)

* PythonPackage: check purelib for libs/headers

* Update error messages too

* Fix functools.reduce argument order

johnwparent committed:
netcdf-c package: fix hdf5 linking on Windows (#42749)

payerle committed:
qb3: add custom libs property (#42275)

alalazo committed:
elpa: remove deprecated versions (#42802)

akimler committed:
matio: add v1.5.26  (#42808)

alalazo committed:
Fix wrong call to a function (#42814)

alalazo committed:
Remove dead code from Environment (#42818)

Environment.concretize_and_add is not used anywhere.

w8jcik committed:
py-argparse-dataclass: add new package (#42494)

* py-argparse-dataclass: add new package

* Remove obvious dependency

alecbcs committed:
py-ansible: add v2.16.3 (#42734)

* py-ansible: add v2.16.3

* Apply suggestions from code review

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* Add specific python version requirements from setup.cfg

* Add additional ranges for py-setuptools

* Update var/spack/repos/builtin/packages/py-ansible/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

w8jcik committed:
py-s3transfer, py-boto3, py-botocore: add new versions (#42741)

* py-s3transfer: add new versions

* py-boto3: add new versions

* py-botocore: add new versions

* py-boto3: correct version ranges

cmelone committed:
py-codespell: add package (#42694)

* py-codespell: add package

* setuptools-scm conflict

confirmed via https://github.com/codespell-project/codespell/issues/3365

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

* reorder dependencies and versions

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

Sinan81 committed:
pdal: fix version range for patch (#42769)

* Update package.py

fix bug.

* Update var/spack/repos/builtin/packages/pdal/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

tgamblin committed:
`py-sympy`: add version 1.12 (#42770)

adamjstewart commented on var/spack/repos/builtin/packages/py-ipyvuetify/package.py:
```diff
@@ -0,0 +1,28 @@
+
+    depends_on("py-jupyter-packaging", type=("build", "run"))
+    depends_on("py-jupyterlab", type=("build", "run"))
+    depends_on("py-pynpm", type=("build", "run"))
```
Missing some deps and versions, some types are incorrect:
```toml
[build-system]
requires = ["jupyter_packaging~=0.7.9", "jupyterlab~=3.0", "setuptools>=40.8.0", "wheel"]
```
```python
    install_requires=[                                                                   
        "ipyvue>=1.7,<2",                                                                
    ],     
```

jeremyfix committed:
py-ipyvue: new package (#42789)

* add spack recipe for ipyvue

* Specify version for ipywidgets

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

heerener committed:
py-find-libpython: new package (#42804)

* py-find-libpython: new package

* Update var/spack/repos/builtin/packages/py-find-libpython/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

jeremyfix committed:
py-reacton: new package (#42794)

* adds the spack recipe for reacton python package

* Fix versions for ipywidgets and typing-extensions

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

jeremyfix committed:
py-jwcrypto: new package (#42783)

* adds the spack recipe for py-jwcrypto

* split long line to fix E501

* Specify versions for py-cryptography and py-typing-extensions

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

drbenmorgan committed:
geant4: new version v11.2.1 (#42822)

marcodelapierre committed:
Update/add packages in the Nextflow ecosystem (#42776)

Signed-off-by: Dr Marco Claudio De La Pierre <marco.delapierre@gmail.com>

alalazo committed:
Cleanup spack.schema (#42815)

* Move spec_list into its own file, instead of __init__.py

* Remove spack.schema.spack

This module was introduced in #33960 It's almost an exact duplicate of
spack.schema.env, and is not used anywhere.

* Fix typo

simonpintarelli committed:
tiled-mm: add v2.3 (#42829)

scheibelp committed:
Alert user to failed concretizations (#42655)

With this change an error message is emitted when the result of concretization 
is in an inconsistent state.

johnwparent committed:
vtk package: support vtk@9 on Windows (#42751)

johnwparent committed:
Windows path handling: change representation for paths with spaces (#42754)

Some builds on Windows break when encountering paths with spaces. This
reencodes some paths in Windows 8.3 filename format (when on Windows):
this serves as an equivalent identifier for the file, but in a form that
does not have spaces.

8.3 filenames are also truncated in length, which could be helpful, but
that is not the primary intended purpose of using this format.

Overall

* nmake/msbuild packages do this generally for the install prefix
* curl/perl require additional modifications (as written now, each package
  may require calls to `windows_sfn` to work when the Spack
  root/install/staging prefixes contain spaces)

Some items for follow-up:

* Spack itself does not create paths with spaces "on top" of whatever
  the user configures or where it is placed (e.g. the Spack root, the
  staging directory, etc.), so it might be possible to edit some of these
  paths once and avoid a proliferation of individual `windows_sfn`
  calls in individual packages.
* This approach may result in the insertion of 8.3-style paths into
  build artifacts (on Windows), handling this may require additional
  bookkeeping (e.g. when relocating).

vicentebolea committed:
proj: apply stdint.h patch in version 8 (#42791)

* proj: apply stdint.h patch in version 8

* Update var/spack/repos/builtin/packages/proj/package.py

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

---------

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

Chrismarsh committed:
netcdf-cxx4: convert to CMake-based build (#42766)

The CMake-based build is anticipated to work in all cases where the
Autotools-based build did, and to address all prior issues with less
maintenance of the package. In detail:

* Fixes #42735 (CMake's find_package helps with linking to proper
  netcdf-c)
* Replaces older Autotools-based build
* All preexisting variants are handled
* Record hdf5 as an explicit dependency (was missing before)
* Add +tests option

Co-authored-by: Chrismarsh <Chrismarsh@users.noreply.github.com>

AlexanderRichert-NOAA committed:
libjpeg-turbo: add v2.1.5, update recipe (#37963)

Co-authored-by: Alec Scott <hi@alecbcs.com>
Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

jeremyfix committed:
Merge branch 'py-ipyvuetify' of github.com:jeremyfix/spack into py-ipyvuetify

jeremyfix committed:
update to version 1.9.0

jeremyfix committed:
specify minimal required python

jeremyfix committed:
add missing py-wheel dependency

jeremyfix committed:
add missing py-setuptools minimal version

jeremyfix committed:
add missing ipyvue project dependency

jeremyfix committed:
specify jupyter-packaging minimal version

jeremyfix committed:
add missing jpuyterlab minimal version

jeremyfix commented:
Sorry for the cluttered list of commits , I rebased my branch but maybe not the right way ; 

If you prefer we reopoen a PR to have a cleaner history , just let me know and I shall reopen one ; 

For the versions @adamjstewart  , I'm not sure how to specify a version exclusion such as :  "jupyter_packaging~=0.7.9", "jupyterlab~=3.0". Hence, I specified it as 

```
depends_on("py-jupyter-packaging@0.7.10:", type=("build", "run"))
depends_on("py-jupyterlab@3.0.1:", type=("build", "run"))
```

jeremyfix commented:
I prefer closing my PR and resubmitting it , I completely messed up the history and I'm not satisfied with this.
